### PR TITLE
more signals

### PR DESCRIPTION
**Pull request by SuperFluffy**

### CODE REVIEW

File: src/bin/conductor.rs [hunk 1]
```diff
@@ -73,22 +97,50 @@ async fn run() -> Result<()> {
...
+    reload_rx: watch::Receiver<()>,
+    stop_rx: watch::Receiver<()>,
+}
+
+fn spawn_signal_handler() -> SignalReceiver {
+    let (stop_tx, stop_rx) = watch::channel(());
+    let (reload_tx, reload_rx) = watch::channel(());
+    tokio::spawn(async move {
+        let mut sighup = signal(SignalKind::hangup()).expect(
+            "setting a SIHGUP listerner should always work on linux; is this running on linux?",
```
Comment (noot) [inline src/bin/conductor.rs:119]:
```suggestion
            "setting a SIGHUP listerner should always work on linux; is this running on linux?",
```

File: src/bin/conductor.rs [hunk 2]
```diff
@@ -52,15 +61,30 @@ async fn run() -> Result<()> {
...
+            _ = stop_rx.changed() => {
+                info!("shutting down conductor");
+                if let Some(e) = driver_tx.send(DriverCommand::Shutdown).err() {
+                    error!("error sending Shutdown command to driver: {}", e);
+                }
+                break;
+            }
+
+            _ = reload_rx.changed() => {
+                info!("reloading is currently not implemented");
```
Comment (noot) [inline src/bin/conductor.rs:79]: what is reloading supposed to do?

File: src/bin/conductor.rs [hunk 3]
```diff
@@ -52,15 +61,30 @@ async fn run() -> Result<()> {
...
     let mut interval = time::interval(Duration::from_secs(3));
 
-    let mut run = true;
-    while run {
-        tokio::select! {
+    loop {
+        select! {
+            // FIXME: The bias should only be on the signal channels. The the two
+            //       handlers should have the same bias.
+            biased;
```
Comment (noot) [inline src/bin/conductor.rs:68]: what is this for? don't see it used anywhere?

File: src/bin/conductor.rs [hunk 4]
```diff
@@ -7,8 +7,12 @@ use figment::{
...
 };
 use log::{error, info};
-use tokio::sync::mpsc;
-use tokio::{signal, time};
+use tokio::time;
+use tokio::{
+    select,
+    signal::unix::{signal, SignalKind},
+    sync::{mpsc, watch},
+};
```
Comment (noot) [inline src/bin/conductor.rs:15]:
```suggestion
use tokio::{
    select,
    signal::unix::{signal, SignalKind},
    sync::{mpsc, watch},
    time,
};
```
super nit lol

File: src/bin/conductor.rs (hunk 1 above)
Comment (steezeburger) [reply src/bin/conductor.rs:119]:
```suggestion
            "setting a SIGHUP listener should always work on linux; is this running on linux?",
```

File: src/bin/conductor.rs [hunk 5]
```diff
@@ -73,22 +97,50 @@ async fn run() -> Result<()> {
...
+
+fn spawn_signal_handler() -> SignalReceiver {
+    let (stop_tx, stop_rx) = watch::channel(());
+    let (reload_tx, reload_rx) = watch::channel(());
+    tokio::spawn(async move {
+        let mut sighup = signal(SignalKind::hangup()).expect(
+            "setting a SIHGUP listerner should always work on linux; is this running on linux?",
+        );
+        let mut sigint = signal(SignalKind::interrupt()).expect(
+            "setting a SIGINT listerner should always work on linux; is this running on linux?",
```
Comment (steezeburger) [inline src/bin/conductor.rs:122]:
```suggestion
            "setting a SIGINT listener should always work on linux; is this running on linux?",
```

File: src/bin/conductor.rs [hunk 6]
```diff
@@ -73,22 +97,50 @@ async fn run() -> Result<()> {
...
+        );
+        let mut sigterm = signal(SignalKind::terminate()).expect(
+            "setting a SIGTERM listerner should always work on linux; is this running on linux?",
```
Comment (steezeburger) [inline src/bin/conductor.rs:125]:
```suggestion
            "setting a SIGTERM listener should always work on linux; is this running on linux?",
```

**Review by noot — APPROVED** [review]

**Pull request merged**